Running xmlformat of all docs

CWehli opened the pull request:
- Adaptation of the xml-formatting example to the changed doc structure.
- xmlformat of all xml-docs, both guide and manual and all languages

CWehli committed:
Adaptation of the xml-formatting example to the changed doc structure

and insert the processing of *.docbook- files to the for-loop.

CWehli committed:
fix malformed xml

CWehli committed:
xmlformat of all xml-docs, both guide and manual and all languages

fellen commented on C/guide/appendixd.xml:
```diff
@@ -33,15 +33,12 @@
-        dropping a new <filename>*.chk</filename> file in this directory. The next time the check
-        printing dialog is opened the new check format will appear in the list of available check
-        formats.
+        </footnote>directory. Users may add check formats at any time (even while &app; is running) simply by dropping
```
Some post processing or an improvement of our config seem required.
`</footnote> directory`
We want a space if a word follows, but none in case of a sentence sign like comma or full stop.

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -44,18 +44,19 @@ Translators:
     <figure pgwide="1">
-      <title>The basic accounts relationships</title>
-      <screenshot id="accts-AccountRelationships">
+      <title>The basic accounts relationships</title><screenshot id="accts-AccountRelationships">
```
Why this change?

fellen replied:
> Why this change?

Ich glaube, figure sollte element-break=1 bekommen

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -44,18 +44,19 @@ Translators:
             <imagedata fileref="figures/basics_AccountRelationships.png"
                        srccredit="Geert Janssens"/>
           </imageobject>
+
```
Why?

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -465,14 +466,14 @@ Translators:
       </para>
 
-      <para>Start with a clean &app; file by selecting <menuchoice>
+      <para>Start with a clean &app; file by selecting<menuchoice>
```
wrong

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -465,14 +466,14 @@ Translators:
           <guimenu>File</guimenu><guimenuitem>New File</guimenuitem>
-        </menuchoice> from the menu. The <guilabel>New Account Hierarchy Setup</guilabel> assistant will start.
-        Press <guibutton>Cancel</guibutton> to close the assistant as we don&rsquo;t want to use one of
+        </menuchoice>from the menu. The <guilabel>New Account Hierarchy Setup</guilabel> assistant will start. Press
```
wrong

fellen reviewed (changes requested):
[Edit]The first 2 commits are very good, but…[/Edit]
After looking over a few files, I think we need to improve the config file
1. exactly 1 linefeed between elements of `<figure>`
2. 1 space before `<menuchoice>`
3. 1 space after `</footnote>, </menuchoice>` … if the next character is no sentence sign. This issue is probably easier solved by find&replace than in the config.

CWehli committed:
xmlformat.conf: further improvements

- set 'element-break' active for 'para', 'figure' and 'mediaobject'
- insert 'titleabbrev'

CWehli commented:
@fellen: I have adopted the improvements from PR #312 and #241 respectively and think that the formatting looks better now.
Unfortunately, I don't have a solution for all the improvements you suggested.

@jralls: if @fellen agrees with these changes, #312 can be closed.

fellen commented on C/guide/appendixd.xml:
```diff
@@ -807,10 +806,10 @@ Blocking_Chars</literallayout>
-                  Use Split Action Field for Number</ulink> in the Book Options section of the &app;
-                  Manual). If number source for the book is specified as anchor-split action,
-                  this field will instead print the transaction number field.
+                  <ulink url="&url-docs-C;help/book-options.html#num-action-book-option"> Use Split
```
Kein Leerzeichen am Beginn des Labels, also
`<ulink url="&url-docs-C;help/book-options.html#num-action-book-option">Use Split`
Es kann sein, daß der Fehler schon vorher da war, aber nicht gesehen worden ist.

fellen reviewed (changes requested):
Hi, just to get some progress: I just saw, we have `revision` twice defined. Please remove it from the inline section. It should be the cause of some strange formating.

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -45,17 +45,20 @@ Translators:
 
     <figure pgwide="1">
       <title>The basic accounts relationships</title>
+
       <screenshot id="accts-AccountRelationships">
         <mediaobject>
           <imageobject role="html">
             <imagedata fileref="figures/basics_AccountRelationships.png"
                        srccredit="Geert Janssens"/>
           </imageobject>
+
```
Jetzt haben wir einen Umbruch zuviel. Da müssen wir wohl woanders noch enen wegnehmen. Low Prio

CWehli replied:
Ist jetzt behoben.

CWehli replied:
Die zusätzliche Leerzeile bekomme ich leider doch nicht weg.
Wenn das Formatierungsprogramm zwei mal gestartet wird, dann ist diese Leerzeile wieder da. Auch wenn es zuvor einmal richtig funktioniert hat.

fellen replied:
Huch, non-deterministisches Verhalten? Sollten wir im Auge behalten und `3.2.  Line Breaks and Indentation` in file:///usr/share/doc/packages/xmlformat/The%20xmlformat%20XML%20Document%20Formatter.html noch mal in Ruhe mit unserer conf abgleichen.

CWehli replied:
Das Dokument habe ich nicht auf der Platte, ich schaue gerade in diese Beschreibung: [https://github.com/someth2say/xmlformat/blob/main/docs.md](https://github.com/someth2say/xmlformat/blob/main/docs.md).

Ich glaube, das Original unter [http://www.kitebird.com/software/xmlformat/](http://www.kitebird.com/software/xmlformat/) gibt es nicht mehr.

fellen replied:
> Das Dokument habe ich nicht auf der Platte, ich schaue gerade in diese Beschreibung: https://github.com/someth2say/xmlformat/blob/main/docs.md.
> 
> Ich glaube, das Original unter http://www.kitebird.com/software/xmlformat/ gibt es nicht mehr.

Ja, der mögliche Nacholger. Du hast auch https://github.com/someth2say/xmlformat/issues/19 gesehen?

fellen commented on C/guide/ch_accts.xml:
```diff
@@ -369,19 +373,23 @@ Translators:
 
       <para>Below are the standard <guilabel>Income</guilabel> and <guilabel>Expense</guilabel> accounts after
         selecting <guilabel>Common Accounts</guilabel> in the assistant for creating a new Account
-        Hierarchy (<menuchoice>
+        Hierarchy (
+        <menuchoice>
           <guimenu>Actions</guimenu><guimenuitem>New Account Hierarchy...</guimenuitem>
-        </menuchoice>).
+        </menuchoice>
+        ).
```
Das dürfte unerwünschte Leerzeichen in der Klammer ergeben, aber macht die Klammer überhaupt Sinn?  Gab es da keinen Dokbook-terminus für?.

CWehli replied:
Wenn im Guide für die Menüeinträge auch die neuen Entities eingearbeitet sind, dann erledigt sich dieses Problem von selber.

fellen commented on util/xmlformat/README:
```diff
@@ -14,10 +14,10 @@ If you installed the package xmlformat you can shorten it to:
 
 To reformat all files of the english guide:
-for i in guide/C/*.xml; do util/xmlformat/xmlformat.pl -f util/xmlformat/xmlformat.conf -i $i ; done
+for i in C/guide/*.xml; do util/xmlformat/xmlformat.pl -f util/xmlformat/xmlformat.conf -i $i ; done
```
Da das Hauptdokument  in `index.dokbook` umbenannt wurde, wird es nicht mehr erfaßt. M.E. die einfachste und sauberste Lösung: Umbennennung der restlichen .xml nach .dokbook (oder aller nach .dbk, was auch eine  offizielle Endung sein soll —erfordert Test.)

CWehli replied:
Das Hauptdokument ist doch recht statisch, das könnten wir doch auch händisch formatieren. Ich würde die einzelnen XML-Dateien am liebsten so lassen.
[edit] ... und 'it' ist ja nicht mehr wirklich aktuell [/edit]

fellen replied:
There was @gjanssens not very consequent. ;-)

Wenn du meinst, dann lass es drüber laufen. `<revision>` wurde von uns geändert [Edit]und [/Edit]wird darin verwendet.

fellen replied:
> Das Hauptdokument ist doch recht statisch

@jralls adds on each release a section to `<revhistory>` and is not happy with the current line breaks.

fellen reviewed (changes requested):
Lesbarkeit erhöht, danke!
Aber siehe Kommentar in util/xmlformat/README

Ein Konflikt ist übrigens vorprogrammiert: https://github.com/Gnucash/gnucash-docs/pull/323/commits
Mal sehen, ob ich die PR vorher reinnehmen kann…

CWehli commented:
> Lesbarkeit erhöht, danke!
> Aber siehe Kommentar in util/xmlformat/README

Welchen Kommentar meinst du?

fellen commented:
> > Lesbarkeit erhöht, danke!
> > Aber siehe Kommentar in util/xmlformat/README
> 
> Welchen Kommentar meinst du?

https://github.com/Gnucash/gnucash-docs/pull/318/files#r1265881425

fellen commented on util/xmlformat/xmlformat.conf:
```diff
@@ -46,16 +46,20 @@ menuchoice keycombo  # because they are more complex
   element-break = 0
 
+# each sub-element begin a new line:
+figure mediaobject
```
[Edit]Schön, aber [/Edit]
mediaobject in Zeile 40 dann löschen.

CWehli replied:
Leider funktioniert die Lösung doch nicht. Bei einem wiederholten Formatierungslauf wird die Leerzeile wieder eingefügt.

Diese Konfigurationsdatei wird sequenziell abgearbeitet kommt. Die Formatierungs-Einträge für ein Element kann weiter hinten in der Datei gezielt überschrieben werden.
So kann anfänglich eine pauschale Formatierung vorgenommen werden, die dann im Nachgang 'feinjustiert' wird.
Mit Zeile 40 wird das `mediaobjekt` als Block formatiert und hier in Zeile 53 dann mit einer neuen Anweisung ergänzt.

fellen replied:
Dann sollten wir unser Layout für die Datei —vielleicht zunächst im Wiki— dokumentieren.

M.E. sollte jedes benutzte Element genau einmal definiert werden, sonst passieren so unschöne Dinge wie mit `<revision>`.
Mein Ansatz war, Elemente mit gleichen Eigenschaften in Definitionsblöcken zusammenzufassen, also bei Änderungen Elemente zwischen Blöcken zu verschieben.

fellen commented:
Ein aktualisierter Durchlauf nach rebase wäre schön wegen commit 24a91f0.

fellen commented:
Die ganzen `index.dokbook` fehlen immer noch. Dan muß man sie wohl auch im `README` erwähnen. ;-)

CWehli commented:
> Die ganzen `index.docbook` fehlen immer noch.

Für die `index.docbook` habe ich die Formatierung für `<revnumber>` in Zeile 49 zusätzlich modifiziert und dann am Ende aus dem 'Inline-Format' entfernt.
Und die Readme-Datei habe ich entsprechend ergänzt.

fellen commented:
OK, despite the result is not perfect we got much progress.
Thanks!
